lib/Service/Logging.cpp

```diff
@@ -70,60 +75,87 @@
  * External Functions
  *****************************************************************************/
 
-void Logging::printHead(const char* filename, int lineNumber, Logging::LogLevel level)
+bool Logging::isEnabled()
 {
-    const char* levelStr = "U";
-
-    switch (level)
-    {
-    case Logging::LOG_LEVEL_FATAL:
-        levelStr = "F";
-        break;
-
-    case Logging::LOG_LEVEL_ERROR:
-        levelStr = "E";
-        break;
-
-    case Logging::LOG_LEVEL_WARNING:
-        levelStr = "W";
-        break;
+    return gIsLogEnabled;
+}
 
-    case Logging::LOG_LEVEL_INFO:
-        levelStr = "I";
-        break;
+void Logging::enable()
+{
+    gIsLogEnabled = true;
+}
 
-    case Logging::LOG_LEVEL_DEBUG:
-        levelStr = "D";
-        break;
+void Logging::disable()
+{
+    gIsLogEnabled = false;
+}
 
-    default:
-        break;
+void Logging::printHead(const char* filename, int lineNumber, Logging::LogLevel level)
+{
+    if (true == isEnabled())
+    {
+        const char* levelStr = "U";
+
+        switch (level)
+        {
+        case Logging::LOG_LEVEL_FATAL:
+            levelStr = "F";
+            break;
+
+        case Logging::LOG_LEVEL_ERROR:
+            levelStr = "E";
+            break;
+
+        case Logging::LOG_LEVEL_WARNING:
+            levelStr = "W";
+            break;
+
+        case Logging::LOG_LEVEL_INFO:
+            levelStr = "I";
+            break;
+
+        case Logging::LOG_LEVEL_DEBUG:
+            levelStr = "D";
+            break;
+
+        default:
+            break;
+        }
+
+        Serial.print(static_cast<uint32_t>(millis()));
+        Serial.print(" ");
+        Serial.print(levelStr);
+        Serial.print(" ");
+        Serial.print(filename);
+        Serial.print("(");

```

Comment:
Would it be better to use the standard format of VSCode for filename and line?
`<filename>:<lineNumber>`
Example:
lib\Service\Logging.h:23

In that way, you can copy & paste the log into VScode snd navigate to it.

Reply:
Its only possible if the LOG_TAG() contains the whole filename, but right now we use short TAGS.
But of course, I change it to have the possibility in general.